core: import equivocated aura and babe blocks

andresilva opened the pull request:
Currently whenever we see an equivocated Aura or BABE block we'll just error and refuse to import it. This means that nodes will stick to whatever block they saw first regardless of what chain ends up getting longer. We should still import the block regardless of the equivocation, the way to deal with the equivocation is to report it to some runtime module (which doesn't exist yet) that will take care of slashing.

bkchr commented on core/consensus/babe/src/lib.rs:
```diff
@@ -586,23 +586,21 @@ fn check_header<B: Block + Sized, C: AuxStore>(
 			if check(&inout, threshold) {
 				match check_equivocation(&client, slot_now, slot_num, header.clone(), signer.clone()) {
```
`check_equivocation(bla).map_err(ToString::to_string)?`

Will remove some writing below in the match.

The same can probably be done for aura.

andresilva commented:
I'm working on fixing the tests (moving them to `substrate-consensus-slots` instead).

andresilva commented:
Moved the existing tests to the slots module, the aura and babe tests were pretty much duplicate anyway so we're not losing anything. I also cleaned up some stuff to remove some cloning.

rphmeier commented on core/consensus/aura/src/lib.rs:
```diff
@@ -500,26 +500,24 @@ fn check_header<C, B: Block, P: Pair>(
+				&header,
+				public,
+			).map_err(|e| e.to_string())? {
+				// NOTE: we'll want to report this equivocation to some
```
I think this note isn't really necessary. Likewise in BABE.
